Documentation/ApiOverview/Events/Events/Linkvalidator/BeforeRecordIsAnalyzedEvent.rst

```diff
@@ -10,6 +10,144 @@ BeforeRecordIsAnalyzedEvent
+        {
+            // do something
+        }
+    }
+
+The listener must then be registered in the extensions :php:`Services.yaml`:
+
+..  code-block:: yaml
+    :caption: EXT:my_extension/Configuration/Services.yaml
+
+    services:
+        _defaults:
+            autowire: true
+            autoconfigure: true
+            public: false
+
+        Vendor\MyExtension\:
+            resource: '../Classes/*'
+            exclude: '../Classes/Domain/Model/*'
+
+        Vendor\MyExtension\EventListener\Linkvalidator\BeforeRecordIsAnalyzedEventListener:
+            tags:
+                - name: event.listener
+                  method: handleEvent
+                  identifier: 'myExtensionBeforeRecordIsAnalyzedEventListener'
+                  event: TYPO3\CMS\Linkvalidator\Event\BeforeRecordIsAnalyzedEvent
+
+For the implementation we need the :php:`BrokenLinkRepository` to register
+additional link errors and the :php:`SoftReferenceParserFactory` so we can
+automatically parse for links. These two classes have to be injected via
+:ref:`dependeny injection <Dependency-Injection>`:
+
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/EventListener/Linkvalidator/BeforeRecordIsAnalyzedEventListener.php
+
+    use TYPO3\CMS\Core\DataHandling\SoftReference\SoftReferenceParserFactory;
+    use TYPO3\CMS\Linkvalidator\Event\BeforeRecordIsAnalyzedEvent;
+    use TYPO3\CMS\Linkvalidator\Repository\BrokenLinkRepository;
+
+    class BeforeRecordIsAnalyzedEventListener
+    {
+        protected BrokenLinkRepository $brokenLinkRepository;
+        protected SoftReferenceParserFactory $softReferenceParserFactory;
+
+        public function __construct(
+            BrokenLinkRepository $brokenLinkRepository,
+            SoftReferenceParserFactory $softReferenceParserFactory
+        ) {
+            $this->brokenLinkRepository = $brokenLinkRepository;
+            $this->softReferenceParserFactory = $softReferenceParserFactory;
+        }
+    }
+
+In this example we only check the field :sql:`bodytext` of new tables:
+
+
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/EventListener/Linkvalidator/BeforeRecordIsAnalyzedEventListener.php
+
+    public function handleEvent(BeforeRecordIsAnalyzedEvent $event): void
+    {
+        $table = $event->getTableName();
+        $forbiddenDomain = 'example.org';
+        if ($table == 'tx_news_domain_model_news') {
+            $results = $event->getResults();
+            $newsItem = $event->getRecord();
+            // a quick check if the forbidden domain is contained at all
+            if (strpos((string) $newsItem['bodytext'], $forbiddenDomain) !== false) {
+                $field = 'bodytext';
+                $conf = $GLOBALS['TCA']['tx_news_domain_model_news']['columns']['bodytext']['config'];
+                // parse the field
+                foreach ($this->softReferenceParserFactory->getParsersBySoftRefParserList($conf['softref'], ['subst']) as $softReferenceParser) {
```

Comment:
Can we split up these one or two of the foreaches into separate methods with speaking names what is done? This would improve readability a lot.